addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/SameOriginMethodExecutionScanRule.java

```diff
@@ -0,0 +1,227 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrules;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeSet;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.htmlparser.jericho.Element;
+import net.htmlparser.jericho.HTMLElementName;
+import net.htmlparser.jericho.Source;
+import net.sf.json.JSONException;
+import net.sf.json.JSONObject;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HtmlParameter;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.zaproxy.addon.commonlib.CommonAlertTag;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+/**
+ * A scan rule to detect Same Origin Method Execution (SOME)
+ * https://www.blackhat.com/docs/eu-14/materials/eu-14-Hayak-Same-Origin-Method-Execution-Exploiting-A-Callback-For-Same-Origin-Policy-Bypass-wp.pdf
+ *
+ * @author Karthik UJ (@5up3r541y4n)
+ */
+public class SameOriginMethodExecutionScanRule extends PluginPassiveScanner {
+
+    /** Prefix for internationalised messages used by this rule */
+    private static final String MESSAGE_PREFIX = "pscanrules.some.";
+
+    private static final int PLUGIN_ID = 69096;
+    private static final Pattern JSONP_PATTERN =
+            Pattern.compile("([\\w$]+?)\\(.*?\\)", Pattern.CASE_INSENSITIVE);
+    private static final ArrayList<String> CALLBACK_PARAMS =
+            new ArrayList<String>() {
+                {
+                    add("callback");
+                    add("target");
+                    add("cb");
+                    add("jsonp");
+                    add("cmd");
+                    add("readyFunction");
+                    add("jsoncallback");
+                }
+            };
+    private static final Map<String, String> ALERT_TAGS =
+            CommonAlertTag.toMap(
+                    CommonAlertTag.OWASP_2021_A04_INSECURE_DESIGN,
+                    CommonAlertTag.OWASP_2017_A03_DATA_EXPOSED);
+
+    private static final Logger LOGGER =
+            LogManager.getLogger(SameOriginMethodExecutionScanRule.class);
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        HttpMessage newReq = msg.cloneRequest();
```

Comment:
You can just analyze msg directly no need to clone it in a passive rule, you aren't changing or sending it.